repositories/src/main/scala/cromwell/pipeline/datastorage/dao/repository/ProjectConfigurationRepository.scala

```diff
@@ -6,20 +6,35 @@ import cromwell.pipeline.datastorage.dto.{ ProjectConfiguration, ProjectConfigur
 
 import scala.concurrent.{ ExecutionContext, Future }
 
-class ProjectConfigurationRepository(repository: DocumentRepository)(implicit ec: ExecutionContext) {
+trait ProjectConfigurationRepository {
 
-  private def upsertConfiguration(projectConfiguration: ProjectConfiguration): Future[Unit] =
-    repository.upsertOne(projectConfiguration, "id", projectConfiguration.id.value)
+  def addConfiguration(projectConfiguration: ProjectConfiguration): Future[Unit]
 
-  def addConfiguration(projectConfiguration: ProjectConfiguration): Future[Unit] =
-    upsertConfiguration(projectConfiguration)
+  def updateConfiguration(projectConfiguration: ProjectConfiguration): Future[Unit]
 
-  def updateConfiguration(projectConfiguration: ProjectConfiguration): Future[Unit] =
-    upsertConfiguration(projectConfiguration)
+  def getById(id: ProjectConfigurationId): Future[Option[ProjectConfiguration]]
 
-  def getById(id: ProjectConfigurationId): Future[Option[ProjectConfiguration]] =
-    repository.getByParam("id", id.value).map(_.headOption)
+  def getAllByProjectId(projectId: ProjectId): Future[Seq[ProjectConfiguration]]
+}
```

Comment:
[minor] Let's make it symmetrical :) If we have line separator between trait header and the first method then why don't we have such separator between the last method and footer? I.e. I think it would be better to change this
```
trait A {

  def foo = ???

  def bar = ???
}

```
to this
```
trait A {

  def foo = ???

  def bar = ???

}

```

Reply:
@myazinn done